### ISSUE TITLE
base64 encoded images are not displayed following "data:" typed as text -- INCLUDES FIX

### ISSUE THREAD

**Issue by ericfreed-pki** [tinymce/tinymce#7737]
**What is the current behavior? Describe the bug**
From a blank document, type "My data:" then either on the same line or the next paragraph insert an image.
It appears to work fine, but the html that will be generated will not open in tinymce correctly the next time.

Now, on page load set the Tinymce content to that html.
```
<p>My data:</p>
<p><img src="data:image/png;base64,abcdefg..." ...></p>
```

When Tinymce receives that valid html, it winds up html escaping everything between the first "data:" and the base64 part, so the editor appears to show raw html.

**Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tiny.cloud](https://fiddle.tiny.cloud/) or similar.**
https://fiddle.tiny.cloud/taiaab

**What is the expected behavior?**
If the text typed is slightly different like `data :` or `data:;` the bug does not happen, the expected behavior is to behave like that.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**
Tested in version 5.  It happens in all browsers.  I'm looking for the fix in version 5

**The Fix**
This PR 7098 touches the same line and a PR for this fix can be modeled after it.  But I'm not familiar with the testing requirements.
https://github.com/tinymce/tinymce/pull/7098

The first line of the `extractBase64DataUris` function has a regex that starts with `/data:[^;]+;`.
If you insert one more character so it starts with `/data:[^;<]+;` that fixes this bug.
The logic there is wherever "data:" is used correctly, the greater than and less than characters will be required to be escaped.  So to prevent crossing a `<` prevents it from selecting into a separate html tag.

To be consistent with that PR 7098, one line in the `parseDataUri` function should get the same edit as well.

**Comment by TheSpyder** [comment]
Thank you for the suggested fix! We are planning a 5.10.4 release shortly after 6.0 is complete so hopefully we can find room for this.

In the meantime, version 6 does avoid the issue by virtue of switching to the browser parsing APIs. `extractBase64DataUris` isn't used anymore.

**Comment by loganallen** [comment]
Any updates on when a minor version for `5` will be released to include a patch for this? I have also come across this issue and would love a resolution.

**Comment by lnewson** [comment]
This has been fixed in TinyMCE 5.10.5 which has now been released to the community.
